### PR TITLE
Updating DiagnosticSource to 5.0.0-RC version

### PR DESCRIPTION
**Pull request by eddynaka**
Fix Issue # .

## Changes
- Updated DiagnosticSource package to latest rc version
- Removed Unsafe package
- Removed System.Memory package
- Removed DiagnosticSource from test projects

### Checklist
- [ ] I ran Unit Tests locally.
- [ ] CHANGELOG.md updated with one line description of the fix, and a link to the original issue if available.

For significant contributions please make sure you have completed the following items:

- [ ] Design discussion issue #
- [ ] Changes in public surface reviewed

The PR will trigger build, unit tests, and functional tests automatically. Please follow [these](https://github.com/Microsoft/ApplicationInsights-dotnet/blob/develop/.github/CONTRIBUTING.md) instructions to build and test locally.

### Notes for authors:
- FxCop and other analyzers will fail the build. To see these errors yourself, compile localy using the Release configuration.

### Notes for reviewers:

- We support [comment build triggers](https://docs.microsoft.com/azure/devops/pipelines/repos/github?view=azure-devops&tabs=yaml#comment-triggers)
  - `/AzurePipelines run` will queue all builds
  - `/AzurePipelines run <pipeline-name>` will queue a specific build

### CODE REVIEW

File: BASE/Test/ServerTelemetryChannel.Test/TelemetryChannel.Nuget.Tests/TelemetryChannel.Nuget.Tests.csproj [hunk 1]
```diff
@@ -23,7 +23,7 @@
     <PackageReference Include="MSTest.TestFramework" Version="2.1.2" />
     <PackageReference Include="CompareNETObjects" Version="4.59.0" />
     <PackageReference Include="Newtonsoft.Json" Version="10.0.3" />
-    <PackageReference Include="System.Diagnostics.DiagnosticSource" Version="4.6.0" />
+    <PackageReference Include="System.Diagnostics.DiagnosticSource" Version="5.0.0-rc.1.20451.14" />
```
Comment (cijothomas) [inline BASE/Test/ServerTelemetryChannel.Test/TelemetryChannel.Nuget.Tests/TelemetryChannel.Nuget.Tests.csproj:26]: can we remove this reference for this proj? the main sdk project should bring it anyway.

Comment (eddynaka) [reply BASE/Test/ServerTelemetryChannel.Test/TelemetryChannel.Nuget.Tests/TelemetryChannel.Nuget.Tests.csproj:26]: done.

Comment (eddynaka) [reply BASE/Test/ServerTelemetryChannel.Test/TelemetryChannel.Nuget.Tests/TelemetryChannel.Nuget.Tests.csproj]: should i do for all test projects?

Comment (TimothyMothra) [reply BASE/Test/ServerTelemetryChannel.Test/TelemetryChannel.Nuget.Tests/TelemetryChannel.Nuget.Tests.csproj]: if possible, yes please. That will make updates easier in the future.

**Comment by TimothyMothra** [comment]
As part of this you should be able to resolve https://github.com/microsoft/ApplicationInsights-dotnet/issues/1790

Please see if you can remove the explicit references to System.Memory v4.5.4

File: LOGGING/src/DiagnosticSourceListener/DiagnosticSourceListener.csproj [hunk 2]
```diff
@@ -22,8 +22,7 @@
 
   <ItemGroup>
     <ProjectReference Include="..\..\..\BASE\src\Microsoft.ApplicationInsights\Microsoft.ApplicationInsights.csproj" />
-    <PackageReference Include="System.Diagnostics.DiagnosticSource" Version="5.0.0-rc.1.20451.14" />
-    <PackageReference Include="System.Memory" Version="4.5.4" /> <!-- This is a temporary explicit dependency. We're waiting on the next version of System.Diagnostics.DiagnosticSource. https://github.com/microsoft/ApplicationInsights-dotnet/issues/1707 -->
+    <PackageReference Include="System.Diagnostics.DiagnosticSource" Version="5.0.0-rc.1.20451.14" /> <!-- This is a temporary explicit dependency. We're waiting on the next version of System.Diagnostics.DiagnosticSource. https://github.com/microsoft/ApplicationInsights-dotnet/issues/1707 -->
```
Comment (TimothyMothra) [inline LOGGING/src/DiagnosticSourceListener/DiagnosticSourceListener.csproj:25]: can remove this comment

File: NETCORE/src/Microsoft.ApplicationInsights.AspNetCore/Microsoft.ApplicationInsights.AspNetCore.csproj [hunk 3]
```diff
@@ -31,7 +31,6 @@
     <ProjectReference Include="..\..\..\WEB\Src\PerformanceCollector\PerformanceCollector\Perf.csproj" />
     <ProjectReference Include="..\..\..\WEB\Src\WindowsServer\WindowsServer\WindowsServer.csproj" />
     <PackageReference Include="System.Diagnostics.DiagnosticSource" Version="5.0.0-rc.1.20451.14" />
-    <PackageReference Include="System.Memory" Version="4.5.4" />
     <!-- This is a temporary explicit dependency. We're waiting on the next version of System.Diagnostics.DiagnosticSource. https://github.com/microsoft/ApplicationInsights-dotnet/issues/1707 -->
```
Comment (TimothyMothra) [inline NETCORE/src/Microsoft.ApplicationInsights.AspNetCore/Microsoft.ApplicationInsights.AspNetCore.csproj:34]: can remove this comment

File: LOGGING/src/DiagnosticSourceListener/DiagnosticSourceListener.csproj (hunk 2 above)
Comment (eddynaka) [reply LOGGING/src/DiagnosticSourceListener/DiagnosticSourceListener.csproj]: done! thanks for reviewing

File: NETCORE/src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs [hunk 4]
```diff
@@ -229,7 +229,9 @@ public void OnHttpRequestInStart(HttpContext httpContext)
                 {
                     // Scenario #3. W3C-TraceParent
                     // We need to ignore the Activity created by Hosting, as it did not take W3CTraceParent into consideration.
+#pragma warning disable CA2000 // Dispose objects before losing scope
                     newActivity = new Activity(ActivityCreatedByHostingDiagnosticListener);
+#pragma warning restore CA2000 // Dispose objects before losing scope
```
Comment (TimothyMothra) [inline NETCORE/src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs:234]: need to include a comment explaining why we're not disposing this Activity

Comment (eddynaka) [reply NETCORE/src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs:235]: Just added. Let me know what you think. Thanks for reviewing!

Comment (cijothomas) [reply NETCORE/src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs:235]: the comment seems incorrect. We dispose it in the OnEnd callback. We get access to this activity, by using Activity.Current.

Comment (eddynaka) [reply NETCORE/src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs:235]: in which file are we disposing the Activity.Current?

Comment (cijothomas) [reply NETCORE/src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs:235]: I actually meant we are stopping the activity, not dispose. But dispose is doing nothing but stop!

Comment (cijothomas) [reply NETCORE/src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs:235]: https://github.com/dotnet/runtime/blob/master/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs#L871-L885

File: WEB/Src/DependencyCollector/DependencyCollector/DependencyTrackingTelemetryModule.cs [hunk 5]
```diff
@@ -282,9 +282,11 @@ protected virtual void Dispose(bool disposing)
         [MethodImpl(MethodImplOptions.NoOptimization | MethodImplOptions.NoInlining)]
         private static void PrepareFirstActivity()
         {
-            var activity = new Activity("Microsoft.ApplicationInights.Init");
-            activity.Start();
-            activity.Stop();
+            using (var activity = new Activity("Microsoft.ApplicationInights.Init"))
+            {
+                activity.Start();
+                activity.Stop();
```
Comment (cijothomas) [inline WEB/Src/DependencyCollector/DependencyCollector/DependencyTrackingTelemetryModule.cs:288]: no need of stop here, as dispose will do it anyway.

File: NETCORE/src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs (hunk 4 above)
Comment (eddynaka) [reply NETCORE/src/Microsoft.ApplicationInsights.AspNetCore/DiagnosticListeners/Implementation/HostingDiagnosticListener.cs:235]: @cijothomas @TimothyMothra , just updated the comments!

**Review by cijothomas — APPROVED** [review]

**Pull request merged**